src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -662,9 +662,9 @@ def get_categories(self):
         Return a product's categories or parent's if there is a parent product.
         """
         if self.is_child:
-            return self.parent.categories
+            return self.parent.categories.filter(is_public=True)
```

Comment:
Please use the `browsable()` queryset instead - i.e., `self.parent.categories.browsable()`, which correctly filters out categories with non-public ancestors (`is_public` alone doesn't).